diff --git a/include/ruby/ruby.h b/include/ruby/ruby.h
index 0a8407a..22dff42 100644
--- a/include/ruby/ruby.h
+++ b/include/ruby/ruby.h
@@ -1207,6 +1207,9 @@ rb_obj_wb_unprotect(VALUE x, const char *filename, int line)
     RGENGC_LOGGING_WB_UNPROTECT((void *)x, filename, line);
 #endif
 
+	(void) filename;
+	(void) line;
+
 #if USE_RGENGC
     /* `x' should be an RVALUE object */
     if (FL_TEST_RAW((x), FL_WB_PROTECTED)) {
@@ -1227,6 +1230,10 @@ rb_obj_written(VALUE a, VALUE oldv, VALUE b, const char *filename, int line)
     RGENGC_LOGGING_OBJ_WRITTEN(a, oldv, b, filename, line);
 #endif
 
+	(void) oldv;
+	(void) filename;
+	(void) line;
+
 #if USE_RGENGC
     /* `a' should be an RVALUE object */
     if (FL_TEST_RAW((a), FL_OLDGEN) &&
